### PR TITLE
Add compression

### PR DESCRIPTION
**Pull request by pinges**
Make use of the native compression implementation when calculating the profitability of transactions.

### CODE REVIEW

File: native/compress/build.gradle [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Hyperledger Besu Contributors
```
Comment (macfarla) [inline native/compress/build.gradle:2]:
```suggestion
 * Copyright Consensys Software Inc.
```

File: native/compress/src/main/java/net/consensys/linea/compress/LibCompress.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Hyperledger Besu Contributors
```
Comment (macfarla) [inline native/compress/src/main/java/net/consensys/linea/compress/LibCompress.java:2]:
```suggestion
 * Copyright Consensys Software Inc.
```

File: native/compress/build.gradle [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+jar {
+    archiveBaseName = 'linea-native-compress'
+    includeEmptyDirs = false
+    manifest {
+        attributes(
+                'Specification-Title': archiveBaseName,
+                'Specification-Version': project.version,
+                'Implementation-Title': archiveBaseName,
+                'Implementation-Version': project.version,
+                'Automatic-Module-Name': 'org.hyperledger.besu.nativelib.compress'
```
Comment (macfarla) [inline native/compress/build.gradle:85]:
```suggestion
                'Automatic-Module-Name': 'org.consensys.nativelib.compress'
```

Comment (macfarla) [reply native/compress/build.gradle:85]: idk if my suggestion is right but don't think we need besu in here

File: native/compress/src/test/java/net/consensys/linea/compress/LibCompressTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Hyperledger Besu Contributors
```
Comment (macfarla) [inline native/compress/src/test/java/net/consensys/linea/compress/LibCompressTest.java:2]:
```suggestion
 * Copyright Consensys Software Inc.
```

**Review by macfarla — COMMENTED** [review]
minor comments on removing references to besu from the copy over

**Review by fab-10 — APPROVED** [review]

**Pull request merged**